zone/lua_client.cpp

```diff
@@ -1211,7 +1211,22 @@ void Lua_Client::AddPVPPoints(uint32 points) {
 
 void Lua_Client::AddCrystals(uint32 radiant, uint32 ebon) {
 	Lua_Safe_Call_Void();
-	self->AddCrystals(radiant, ebon);
+
+	if (ebon != 0) {
+		if (ebon > 0) {
+			self->AddEbonCrystals(ebon);
+		} else {
+			self->RemoveEbonCrystals(ebon);
+		}
+	}
+
+	if (radiant != 0) {
+		if (radiant > 0) {
+			self->AddRadiantCrystals(radiant);
+		} else {
```

Comment:
Instead of `else {` 'ing - we could return early in the first case and fall through

```cpp
	if (radiant != 0) {
		if (radiant > 0) {
			self->AddRadiantCrystals(radiant);
			return;
		}
		self->RemoveRadiantCrystals(radiant);
	}
```